File: 3.Platform_Solutions/AlgoExpert/Arrays/04-Tournament-Winner/Python/solution.py
def tournamentWinner(competitions, results):
    # Write your code here.
    winner = None
    winning_score = 0
    team_scores = {}

    for i in range(len(competitions)):
        home_team = competitions[i][0]
        away_team = competitions[i][1]

        team_scores.setdefault(home_team, 0)
        team_scores.setdefault(away_team, 0)

        if results[i] == 1:
            team_scores[home_team] = team_scores[home_team] + 3
        else:
            team_scores[away_team] = team_scores[away_team] + 3

    for key, value in team_scores.items():
        if winner == None or value > winning_score:
            winner = key
            winning_score = value
            
    return winner

File: 3.Platform_Solutions/AlgoExpert/Arrays/04-Tournament-Winner/Java/Solution.java
import java.util.*;

class Solution {

  public String tournamentWinner( ArrayList<ArrayList<String>> competitions, ArrayList<Integer> results) {
    // Write your code here.
		String winner = null;
		int winningScore = 0;
		Map<String, Integer> teamScores = new HashMap<>();
		
		for(int i = 0; i < competitions.size(); i++) {
			String homeTeam = competitions.get(i).get(0);
			String awayTeam = competitions.get(i).get(1);
			
			teamScores.putIfAbsent(homeTeam, 0);
			teamScores.putIfAbsent(awayTeam, 0);
			
			if (results.get(i) == 1) {
				teamScores.put(homeTeam, teamScores.get(homeTeam) + 3);
			} else {
				teamScores.put(awayTeam, teamScores.get(awayTeam) + 3);
			}
		}
		
		for (Map.Entry<String, Integer> entry : teamScores.entrySet()) {
			if (winner == null) {
				winner = entry.getKey();
				winningScore = entry.getValue();
			} 
			else if (entry.getValue() > winningScore) {
				winner = entry.getKey();
				winningScore = entry.getValue();
			}
		}
		
    return winner;
  }
}
